homeassistant/components/nws/weather.py

```diff
@@ -205,23 +227,23 @@ def condition(self):
         return None
 
     @property
-    def native_visibility(self):
+    def native_visibility(self) -> int | None:
         """Return visibility."""
         if self.observation:
             return self.observation.get("visibility")
         return None
 
     @property
-    def native_visibility_unit(self):
+    def native_visibility_unit(self) -> str:
         """Return visibility unit."""
         return UnitOfLength.METERS
 
     @property
-    def forecast(self):
+    def forecast(self) -> list[Forecast] | None:
```

Comment:
This line returns type `Forecast` inside the list, but you also created `NWSForecast` which is used for `forecast`.  This seems inconsistent to me.

Reply:
Putting this back into a thread:

> Yes, sorry my python isn't good enough, but from what I gathered, the weather base class is expecting a Forecast type. But this getter is returning extra properties, which isn't allowed on Forecast since it's a TypedDict. The only way I can get it to pass is to extend Forecast for the internal accumulator, but return the base class in the function. Let me
know if there's a better way.

The current implementation passes the type check. I'm ok keeping the current implementation, or removing `NWSForecast` entirely (it is only for typing).  If we keep the current implementation, it would be good to comment that the `NWSForecast` object is only for typing.  We could also consider explicitly requiring that it is used for typing only using `typing.TYPE_CHECKING`.

Reply:
I'll mark the object as TYPE_CHECKING. If it were my own project, I'd probably remove the object entirely, but maybe for a future dev it'd be nice to have the hints. I'll push an update now.